Use the official _BSD_SOURCE feature test macro instead of the meaningless
HAVE_STRSEP macro in order to detect the availability of strsep().

This allows toolchains supporting strsep() to use it instead of the custom
implementation from dhcpdump, which also avoids the following error with some
toolchains:

	In file included from dhcpdump.c:30:0:
	dhcpdump.c: At top level:
	strsep.c:65:23: error: register name not specified for ‘delim’
	  register const char *delim;
	                       ^

Signed-off-by: Benoît Thébaudeau <benoit.thebaudeau@advansee.com>

diff -Nrdup dhcpdump-1.8.orig/dhcpdump.c dhcpdump-1.8/dhcpdump.c
--- dhcpdump-1.8.orig/dhcpdump.c	2008-06-24 05:26:52.000000000 +0200
+++ dhcpdump-1.8/dhcpdump.c	2011-05-31 19:22:15.987388498 +0200
@@ -26,7 +26,7 @@
 #include <regex.h>
 #include "dhcp_options.h"
 
-#ifndef HAVE_STRSEP
+#ifndef _BSD_SOURCE
 #include "strsep.c"
 #endif
 
